### PR TITLE
Feature/flow test

### PR DESCRIPTION
**Pull request by DrTtnk**
Firsts thighs after you read the changes, try the two docker commands in the README and tell me if they works

### CODE REVIEW

File: experiment_buddy/experiment_buddy.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def _ask_experiment_id(cluster, sweep):
         experiment_id = tkinter.simpledialog.askstring(title, "experiment_id")
         root.destroy()
     except:
-        experiment_id = input(f"Running on {title} \ndescribe your experiment (experiment_id):\n")
+        experiment_id = input(f"Running on {title} 😀\ndescribe your experiment (experiment_id):\n")
```
Comment (manuel-delverme) [inline experiment_buddy/experiment_buddy.py:222]: nofunallowed.jpg
i'm afraid it will break some shells :(

File: test_flow.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+# 1 - Create a new virtual env
+virtualenv buddy-env && source ./buddy-env/bin/activate
+
+# 2 - Clone [or create] your project, in this case I'm using a basic mnist_classifier
+git clone -b feature/testing git@github.com:DrTtnk/examples.git
+cd examples
+
+# 3 - Install the dependencies
+pip install -q -e git+https://github.com/ministry-of-silly-code/experiment_buddy.git@feature/flow_test#egg=experiment_buddy # ToDo temporary branch for test, it will be from master when ready
```
Comment (manuel-delverme) [inline test_flow.sh:21]: @feature?

File: test_flow.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+set -e
+
+# Setup phase - Set the correct user.email and user.name and activate the ssh agent
+eval "$(ssh-agent)" && ssh-add
+git config --global user.email "$GIT_MAIL"
+git config --global user.name "$GIT_NAME"
+
+# Setup phase - Also remember to set your $WANDB_API_KEY (More info in the readme)
+[[ -z "$WANDB_API_KEY" ]] && (echo set \"export WANDB_API_KEY=[your-wandb-key]\" which can be found here: https://wandb.ai/settings; exit;)
```
Comment (manuel-delverme) [inline test_flow.sh:11]: the key is in .netrc it's not alwys necessary to explicitly give it.
is WANDB_API_KEY always set?

Comment (manuel-delverme) [reply test_flow.sh:11]: ok wait. this is all the docker test right?
if so put them hidden in a subfolder.

File: experiment_buddy/experiment_buddy.py (hunk 1 above)
Comment (DrTtnk) [reply experiment_buddy/experiment_buddy.py:222]: It doesn't, but this was committed by mistake, reverting

File: test_flow.sh (hunk 2 above)
Comment (DrTtnk) [reply test_flow.sh:21]: Git flow

File: README.md [hunk 4]
```diff
@@ -28,3 +59,24 @@ Example: (Updated)
...
+
+## For the tester
+
+#### Local flow
+```shell
+docker run -v ~/.ssh:/root/.ssh --rm -ti \
+           -e WANDB_API_KEY=$WANDB_API_KEY \
+           -e GIT_MAIL=$(git config user.email) \
+           -e GIT_NAME=$(git config user.name) \
+           -u root:$(id -u $USER) $(docker build -f ./Dockerfile-flow -q .)
```
Comment (d3sm0) [inline README.md:71]: `docker build -q` is too quite here. If the purpose is to test, I think verbose is more user friendly.

Comment (DrTtnk) [reply README.md:71]: Yes and no, `Dockerfile-flow` is a bare-bone python image with just the needed packages preinstalled to speed up the pip installation, nothing else, the whole test is made during the run of the container, there you have all the needed logs. This command overcomes the absence of a `--build` option during the run and reduce the startup routine to a single copy&paste

A possible alternative would be to split the build and the run in two separate commands and remove the `-q`, but, given that the build output has no purpose other than telling that it has downloaded a python image, I'd prefer to keep the noise as low as possible

File: README.md [hunk 5]
```diff
@@ -28,3 +59,24 @@ Example: (Updated)
...
 1. experiment-buddy will add tagged commits to a dangling branch 
 2. Supports: Unix based OS
+
+## For the tester
+
+#### Local flow
+```shell
+docker run -v ~/.ssh:/root/.ssh --rm -ti \
+           -e WANDB_API_KEY=$WANDB_API_KEY \
+           -e GIT_MAIL=$(git config user.email) \
```
Comment (d3sm0) [inline README.md:69]: #49

**Review by d3sm0 — APPROVED** [review]

File: test_flow.sh (hunk 3 above)
Comment (DrTtnk) [reply test_flow.sh]: moved

**Pull request merged**